File: tests/fixtures/sample_csharp/SampleModule.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace MyApp.Models
{
    /// <summary>
    /// Represents a user in the system.
    /// </summary>
    public class User : BaseEntity, ISerializable
    {
        public string Name { get; set; }
        public int Age { get; set; }

        public User(string name, int age)
        {
            Name = name;
            Age = age;
        }

        /// <summary>
        /// Validates the user data.
        /// </summary>
        public bool Validate()
        {
            return !string.IsNullOrEmpty(Name) && Age > 0;
        }

        public static User Create(string name)
        {
            return new User(name, 0);
        }

        private void LogAction(string action)
        {
            Console.WriteLine(action);
        }
    }

    public interface IProcessor<T>
    {
        void Process(T item);
        bool CanProcess(T item);
    }

    public class UserProcessor : IProcessor<User>
    {
        public void Process(User item)
        {
            item.Validate();
        }

        public bool CanProcess(User item)
        {
            return item != null;
        }
    }

    public enum UserRole
    {
        Admin,
        Editor,
        Viewer
    }

    public struct Point
    {
        public double X;
        public double Y;

        public double DistanceTo(Point other)
        {
            var dx = X - other.X;
            var dy = Y - other.Y;
            return Math.Sqrt(dx * dx + dy * dy);
        }
    }

    public record UserDto(string Name, int Age);

    public abstract class Repository<T> where T : class
    {
        public abstract T GetById(int id);
        public abstract void Save(T entity);
    }

    public static class Extensions
    {
        public static string ToUpperCase(this string s)
        {
            return s.ToUpper();
        }
    }
}

File: tests/fixtures/sample_java/SampleModule.java
package com.example.app;

import java.util.List;
import java.util.Optional;

/**
 * Represents a user in the system.
 */
public class User extends BaseEntity implements Serializable {
    private String name;
    private int age;
    public static final int MAX_AGE = 150;

    public User(String name, int age) {
        this.name = name;
        this.age = age;
    }

    /**
     * Returns the user's name.
     */
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public static User create(String name, int age) {
        return new User(name, age);
    }
}

public interface UserService {
    Optional<User> findById(long id);
    List<User> findAll();
}

public enum Role {
    ADMIN,
    USER,
    GUEST;
}

public record Point(int x, int y) {}

public abstract class BaseEntity {
    protected long id;

    public abstract long getId();
}
